docs/get-started-dapr-secret-store-configuration-provider-aspnet-core.md

```diff
@@ -0,0 +1,352 @@
+
+```csharp
+return Host.CreateDefaultBuilder(args)
+    .ConfigureServices((services) =>
+    {
+        // Add the DaprClient to DI.
+        services.AddSingleton<DaprClient>(client);
+    })
+    .ConfigureAppConfiguration((configBuilder) =>
+    {
+        // Create descriptors for the secrets you want to rerieve from the Dapr Secret Store.
+        var secretDescriptors = new DaprSecretDescriptor[]
+        {
+            new DaprSecretDescriptor(
+                "super-secret",
+                new Dictionary<string, string>(){ { "namespace", "default" } }
+            )
+        };
+
+        // Add the secret store Configuration Provider to the configuration builder.
+        configBuilder.AddDaprSecretStore("kubernetes", secretDescriptors, client);
+    })
+    .ConfigureWebHostDefaults(webBuilder =>
+    {
+        webBuilder.UseStartup<Startup>();
+    });
+```
+
+## Using the Dapr Secret Store Configuration Provider in ASP.NET Core
+
+### 1. Create project and add dependencies
+
+Firts create a web project and add the Dapr.Extensions.Configuration nuget package:
+
+```bash
+# Create an empty web project 
+dotnet new web -o SecretStoreConfigurationProviderSample
+
+cd SecretStoreConfigurationProviderSample
+
+# Add Dapr.Extensions.Configuration nuget package
+dotnet add package Dapr.Extensions.Configuration -v 0.6.0-preview01
+```
+
+### 2. Update Program.cs
+
+Update Program.cs with the following contents: 
+
+```csharp
+namespace SecretStoreConfigurationProviderSample
+{
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Extensions.Hosting;
+    using Dapr.Client;
+    using Dapr.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using System.Net.Sockets;
+    using System;
+    using System.Net;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Secret Store Configuration Provider Sample.
+    /// </summary>
+    public class Program
+    {
+        const string localhost = "127.0.0.1";
+
+        static string daprPort => Environment.GetEnvironmentVariable("DAPR_GRPC_PORT") ?? "50001";
+
+        /// <summary>
+        /// Main for Secret Store Configuration Provider Sample.
+        /// </summary>
+        /// <param name="args">Arguments.</param>
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        /// <summary>
+        /// Creates WebHost Builder.
+        /// </summary>
+        /// <param name="args">Arguments.</param>
+        /// <returns>Returns IHostbuilder.</returns>
+        public static IHostBuilder CreateHostBuilder(string[] args)
+        {
+            // Wait for Dapr side car to be up and running
+            WaitForDapr();
+
+            // Create Dapr Client
+            var client = new DaprClientBuilder()
+                .UseEndpoint($"https://{localhost}:{daprPort}")
```

Comment:
we are planning to release 0.6.0 next week, and I will remove the workaround for tls in the sdk, since this sample is going in the master branch can we remove the endpoint from here. Also would you be fine, if we merge it by mid-next week?

Reply:
it's ready!